### DISCUSSION TITLE
Guidance for adding new language support for Slint

### DISCUSSION THREAD

**Discussion by bmitc** [slint-ui/slint#6991] [Q&A]
Hello! While I am interested in trying out Slint using Python and/or Rust, I am also interested in using it from other languages/VMs.

Is there general guidance for how to go about adding Slint support to another language? Does a compiler for the Slint declarative language that generates code in the target language need to be built? Do bindings to some C library need to be made? Any ordering of steps needed would be helpful!

If it all helps, the languages (and their associated VMs) I am interested in adding support for are F# (.NET/CLR) and Elixir (Erlang/BEAM). I am familiar with building bindings to C libraries, such as GLFW, inany languages and with Qt Quick using QML and Qt for Python.

Thank you!

**Comment by bmitc** [comment]
Also, I am aware of https://github.com/slint-ui/slint/discussions/3550, but I would like to understand things from the ground up and find it is often easier to build things up from scratch once that understanding comes into place. Plus, I am not aware of any attempts for Erlang or Elixir.

**Comment by ogoffart** [comment]
Thank for your interest.

There is currently no such guidance. 

> Does a compiler for the Slint declarative language that generates code in the target language need to be built?

That depends on the language. If this is a compiled language, it would make sense. Or if there are types to generate to ease linting.


But there are two approach
 1. make binding for the slint-interpreter crate
 2. have deeper integration with a compiler to the target language.
